src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java

```diff
@@ -68,11 +69,14 @@ public static long lastModified(InternalMessages.StopEstimate estimate) {
     }
 
     public static GtfsRealtime.TripUpdate newTripUpdate(InternalMessages.StopEstimate estimate) {
-        final String routeName = reformatRouteName(estimate.getTripInfo().getRouteId());
         final int direction = PubtransFactory.joreDirectionToGtfsDirection(estimate.getTripInfo().getDirectionId());
+        String routeId = estimate.getTripInfo().getRouteId();
+        if (!ProcessorUtils.isMetroRoute(routeId)) {
+            routeId = reformatRouteName(routeId);
```

Comment:
I guess the method name should be reformatRouteId

Reply:
Btw why metroRouteId doesn't need reformatting? I also wonder why do you need to remove the last character if it is a number?

Reply:
It is a route variant. I don't know why it is removed.